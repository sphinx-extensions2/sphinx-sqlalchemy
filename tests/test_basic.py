"""Basic tests"""
import os.path
import sys

from sphinx_pytest.plugin import CreateDoctree


def test_basic(sphinx_doctree_no_tr: CreateDoctree, snapshot):
    """Basic test"""
    sys.path.insert(0, os.path.join(os.path.dirname(__file__), "modules"))
    sphinx_doctree_no_tr.set_conf({"extensions": ["sphinx_sqlalchemy"]})
    result = sphinx_doctree_no_tr(".. sqla-model:: module1.TestUser")
    assert not result.warnings
    assert "\n".join([li.rstrip() for li in result.pformat().splitlines()]) == snapshot
