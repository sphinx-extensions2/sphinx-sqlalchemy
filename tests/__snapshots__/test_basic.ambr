# serializer version: 1
# name: test_basic
  '''
  <document source="<src>/index.rst">
      <definition_list classes="simple sqla">
          <definition_list_item>
              <term>
                  module1.TestUser
                   (
                  <emphasis>
                      dbusers
                  )
              <definition>
                  <paragraph>
                      A
                      <literal>
                          user
                      .
                  <rubric>
                      Columns:
                  <table align="left" classes="colwidths-auto">
                      <tgroup cols="3">
                          <colspec>
                          <colspec>
                          <colspec>
                          <tbody>
                              <row>
                                  <entry>
                                      <paragraph>
                                          <emphasis>
                                              pk*
                                  <entry>
                                      <paragraph>
                                          INTEGER
                                  <entry>
                                      <paragraph>
                              <row>
                                  <entry>
                                      <paragraph>
                                          first_name
                                  <entry>
                                      <paragraph>
                                          VARCHAR?
                                  <entry>
                                      <paragraph>
                                          The name of the user.
                              <row>
                                  <entry>
                                      <paragraph>
                                          last_name
                                  <entry>
                                      <paragraph>
                                          VARCHAR(255)?
                                  <entry>
                                      <paragraph>
                                          The surname of the user.
                              <row>
                                  <entry>
                                      <paragraph>
                                          dob
                                  <entry>
                                      <paragraph>
                                          DATE
                                  <entry>
                                      <paragraph>
                                          The date of birth.
                  <rubric>
                      Constraints:
                  <bullet_list>
                      <list_item>
                          <paragraph>
                              PRIMARY KEY (pk)
                      <list_item>
                          <paragraph>
                              UNIQUE (first_name, last_name)
  '''
# ---
